### ISSUE TITLE
Control-plane provisioning doesn't complete when used with a physical machine

### ISSUE THREAD

**Issue by shachard** [metal3-io/metal3-dev-env#1028]
Hi,
I deployed the metal3-dev-env setup and tried to replace the VMs with physical machines (by adding my physical network interface to the "provisioning" bridge).
Everything seemed to work fine until I provisioned the control-plane, but then I got "provisioned" state on the baremetalhost:

NAMESPACE   NAME     STATE         CONSUMER                         ONLINE   ERROR   AGE
metal3            node-1   **provisioned**  test1-controlplane-tb79x   true                       19h

and "provisioning" state on the corresponding machine object:

NAMESPACE   NAME          CLUSTER   NODENAME   PROVIDERID   PHASE          AGE   VERSION
metal3             test1-jgn55  test1                                                       **Provisioning** 18h    v1.23.5

When I looked at the **machine** object I saw:

Status:
  Bootstrap Ready:  true
  Conditions:
    Last Transition Time:  2022-06-22T13:47:46Z
    Message:               1 of 2 completed
    Reason:                **SettingProviderIDOnNodeFailed**
    Severity:              **Error**
    Status:                False
    Type:                  Ready
    Last Transition Time:  2022-06-22T13:39:26Z
    Status:                True
    Type:                  BootstrapReady
    Last Transition Time:  2022-06-22T13:47:46Z
    Message:               requeue in: 30s
    Reason:                **SettingProviderIDOnNodeFailed**
    Severity:              **Error**
    Status:                False
    Type:                  InfrastructureReady
    Last Transition Time:  2022-06-22T13:39:26Z
    Reason:                **WaitingForNodeRef**
    Severity:              Info
    Status:                False
    Type:                  NodeHealthy
  Last Updated:            2022-06-22T13:39:26Z
  Observed Generation:     2
  Phase:                   **Provisioning**

I looked at the log of the **baremetal-operator-controller-manager** pod and saw:

{"level":"error","ts":1655905167.0297885,"logger":"controller.baremetalhost","msg":"Reconciler error","reconciler group":"metal3.io","reconciler kind":"BareMetalHost","name":"node-1","namespace":"metal3","error":"failed to save host status after \"available\": Operation cannot be fulfilled on baremetalhosts.metal3.io \"node-1\": the object has been modified; please apply your changes to the latest version and try again","errorVerbose":"Operation cannot be fulfilled on baremetalhosts.metal3.io \"node-1\": the object has been modified; please apply your changes to the latest version and try again\nfailed to save host status after \"available\"\ngithub.com/metal3-io/baremetal-operator/controllers/metal3%2eio.(*BareMetalHostReconciler).Reconcile\n\t/workspace/controllers/metal3.io/baremetalhost_controller.go:302\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.2/pkg/internal/controller/controller.go:114\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.2/pkg/internal/controller/controller.go:311\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.2/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.2/pkg/internal/controller/controller.go:227\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1581","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.2/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.2/pkg/internal/controller/controller.go:227"}

The same setup worked for me with the VMs.

Thanks,
Shachar

**Comment by shachard** [comment]
Hi,

I attached the capm3-controller-manager log.
[capm3-controller-manager.log](https://github.com/metal3-io/metal3-dev-env/files/9057411/capm3-controller-manager.log)

If I understand correctly, it is trying to verify that the master node is up and running by accessing 192.168.111.249, is that right?

The problem might be that the network interfaces on my physical node are different than the interfaces on the emulated nodes (VMs) - 
I saw that the VMs have enp1s0 and enp2s0 and my real node inspected with eno0,eno1,eno2,eno3,ens2f0 and ens2f1, from which ens2f0 and ens2f1 are the interface I would like to use...

Is there a way to configure this?

Thanks,
Shachar
